butterknife/src/main/java/butterknife/internal/ButterKnifeProcessor.java

```diff
@@ -667,15 +670,37 @@ private void parseResourceStringArray(Element element,
 
     // Assemble information on the field.
     String name = element.getSimpleName().toString();
-    int id = element.getAnnotation(BindStringArray.class).value();
+    int id = element.getAnnotation(BindArray.class).value();
 
     BindingClass bindingClass = getOrCreateTargetClass(targetClassMap, enclosingElement);
-    FieldResourceBinding binding = new FieldResourceBinding(id, name, "getStringArray");
+    FieldResourceBinding binding = new FieldResourceBinding(id, name, methodName);
     bindingClass.addResource(binding);
 
     erasedTargetNames.add(enclosingElement.toString());
   }
 
+  /**
+   * Returns a method name from the {@link android.content.res.Resources} class for array resource
+   * binding, null if the element type is not supported.
+   */
+  private String getArrayResourceMethodName(Element element) {
+    TypeMirror typeMirror = element.asType();
+    if (TypeKind.ARRAY.equals(typeMirror.getKind())) {
+      ArrayType arrayType = (ArrayType) typeMirror;
+      String componentType = arrayType.getComponentType().toString();
+      if ("java.lang.String".equals(componentType)) {
+        return "getStringArray";
+      } else if ("int".equals(componentType)) {
+        return "getIntArray";
+      } else if ("java.lang.CharSequence".equals(componentType)) {
+        return "getTextArray";
+      }
+    } else if (TYPED_ARRAY_TYPE.equals(typeMirror.toString())) {
```

Comment:
I'd prefer if this method was written as:
```java
if (typed array) {
  return "obtain..";
}
if (array) {
 ...
}
return null;
```
to avoid the so-called dangling else which requires me to backtrack and figure out what it's behavior is.